### PR TITLE
[p5.js 2.0] Color module rewrite

### PR DESCRIPTION
**Pull request by limzykenneth**
Rewrite color module based on findings in #7374 and enable HDR color support in 2D canvas.

### CODE REVIEW

**Comment by limzykenneth** [comment]
@davepagurek @ksen0 Here's an example HDR sketch with the HDR renderer: https://editor.p5js.org/limzykenneth/sketches/LB2vJFdj4

You'd need a HDR display to be able to properly view the sketch (eg. iPhone or newer Macs on Safari or Chrome).

**Comment by limzykenneth** [comment]
Almost done with color.js based implementation. Created another sketch to show comparison of hue wheel of HSL, LCH, and okLCH colors: https://editor.p5js.org/limzykenneth/sketches/66VHrGzCo

okLCH is noticeably better at accurate hue angles, eg. see where the primaries RGB are located on the wheel and the distribution of their intermediate colors.

File: src/shape/custom_shapes.js [hunk 1]
```diff
@@ -650,17 +650,15 @@ class Shape {
...
         queue.shift(),
         queue.shift()
       ];
-      return new Color(
-        array.map((v, i) => v * original.maxes[original.mode][i]),
-        original.mode,
-        original.maxes
-      );
+      return new Color(original);
     }
```
Comment (limzykenneth) [inline src/shape/custom_shapes.js:662]: @davepagurek I'm not sure what this is meant to do but the signature of `Color` constructor is now relatively simplified. If the intention is to make a copy, pass the color object to the constructor will do so.

Comment (limzykenneth) [reply src/shape/custom_shapes.js:662]: I'm guessing the visual test is failing because of this change.

Comment (davepagurek) [reply src/shape/custom_shapes.js:662]: oh yeah, so this is part of some code that interpolates values. It's kind of doing a lerpColor, but the color is just one of multiple properties (that vary dynamically) that get interpolated, so it serializes everything to an array of numbers, lerps all of them, then turns them back into their constituent parts.

So this code was previously trying to make a new color with the maxes as the original, but with the lerped outputs from before. What would be the best way to go about this now?

Comment (davepagurek) [reply src/shape/custom_shapes.js]: that looks right to me, thanks!

**Pull request merged**